src/router.js

```diff
@@ -286,28 +285,17 @@ export class Router {
   * @param config The moduleId, or a function that selects the moduleId, or a [[RouteConfig]].
   */
   handleUnknownRoutes(config?: string|Function|RouteConfig): void {
-    let callback = instruction => new Promise((resolve, reject) => {
-      function done(inst) {
-        inst = inst || instruction;
-        inst.config.route = inst.params.path;
-        resolve(inst);
-      }
-
-      if (!config) {
-        instruction.config.moduleId = instruction.fragment;
```

Comment:
This behavior seems highly specialized, and likely to cause very confusing problems if null or undefined are accidentally passed. It's trivial for an application to implement if desired, so I think it's better not to include it by default.